### ISSUE TITLE
Fail with error if user attempts to uninstall NVDA while it is running

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2890]
**Reported by elliott94 on 2012-12-22 23:40**
Currently, if eSpeak is set as the default synth during an uninstall, it stops producing output after the process has started, i.e. the Uninstall button is pressed. However, a dialog is presented at the end of the uninstall which prompts the user as to whether they want to reboot now or later. However, this prompt doesn't get any feedback. This doesn't occur if another synth is selected.

**Comment by bhavyashah** [comment]
@jcsteh Could you please modify the ticket title accordingly to reflect our mostly agreed decision that a warning/error should be presented to mitigate any potential confusion as noted in yours and Mick's comments?

**Comment by Adriani90** [comment]
@elliott94 can you still reproduce this with NVDA 2018.4 Beta 1?

**Comment by jcsteh** [comment]
Even if the original issue can't be reproduced (and I suspect it can), the idea here is still valid; i.e. fail with an error if the user attempts to uninstall while NVDA is running.See https://github.com/nvaccess/nvda/issues/2890#issuecomment-155303469 and https://github.com/nvaccess/nvda/issues/2890#issuecomment-155303470.

**Comment by leonardder** [comment]
It should be not too difficult to do this, see https://nsis.sourceforge.io/Check_whether_your_application_is_running
I'd go for the mutex approach. SInce NSIS isn't very hard to dive into and the reference is pretty neat, marking this as good for new dev.

**Comment by Adriani90** [comment]
cc: @seanbudd

**Comment by seanbudd** [comment]
@Adriani90 could you please explain the tag?

**Comment by Adriani90** [comment]
Ah sorry, I just thought you might want to give it a priority. It still affects users, especially when the dialog after uninstall is not spoken at all.
